facebook/react
# Always use event capturing. Remove media events.

**What**

I finally had a chance to circle back on what I learned investigating event listener attachment. 

In short, we distinguished between captured and bubbled events, however I believe this is not necessary as of IE9+. Captured do not appear to exhibit the same edge cases as bubbled events, allowing us to simplify event code.

**Why do we want to do this?**

I think we can probably drastically reduce the code for eagerly attaching listeners for events that do not bubble. So far, all events that do not bubble appear to capture on parents elements. 

Longer term, [I wonder if we can remove the `.bind()` when attaching listeners](https://github.com/facebook/react/blob/master/packages/react-dom/src/events/ReactDOMEventListener.js#L180), and maybe even remove top level types.

**More info**

This commit changes event subscription such that all events are attached with capturing. This prevents the need to eagerly attach media event listeners to video, audio, and source tags, and simplifies the event subscription process.

I think we can probably extend this to more event types, maybe [most/all eager event attachments](https://github.com/facebook/react/blob/master/packages/react-dom/src/client/ReactDOMFiberComponent.js#L451-L516), but that will require substantial testing. Until that time, this commit lays the foundation for that work by demonstrating that capturing is applicable in cases where bubbling does not occur.

**Test plan**

1. Verify fixtures on https://react-capture-all-events.surge.sh/media-events
2. Verify fixtures on https://react-capture-all-events.surge.sh/input-change-events
3. Verify fixtures on https://react-capture-all-events.surge.sh/pointer-events

I believe these to be the key event types to test, however this change definitely affects all event listeners.

**I've tested so far in:**

- Internet Explorer: 9, 11
- Edge 16
- Firefox: 49, 60
- Chrome 47, 66
- Safari 9, 11.1
